db/commitlog/commitlog.cc

```diff
@@ -1675,6 +1679,65 @@ void db::commitlog::segment_manager::flush_segments(uint64_t size_to_remove) {
     totals.bytes_flush_requested += flushing;
 }
 
+void db::commitlog::segment_manager::check_data_age() {
+    if (!cfg.commitlog_data_max_lifetime_in_seconds) {
+        return;
+    }
+
+    auto now = gc_clock::now();
+
+    std::unordered_set<cf_id_type> ids;
+    std::optional<replay_position> high;
+
+    for (auto& s : _segments) {
+        auto rp = replay_position(s->_desc.id, db::position_type(s->size_on_disk()));
+        if (rp <= _flush_position) {
+            // already requested.
+            continue;
+        }
+
+        for (auto [id, lo] : s->_cf_min_time) {
+            // Ignore CF:s already released.
+            if (!s->_cf_dirty.count(id)) {
+                continue;
+            }
+
+            auto time_here = std::chrono::duration_cast<std::chrono::seconds>(now - lo).count();
+            if (time_here >= *cfg.commitlog_data_max_lifetime_in_seconds) {
+                for (auto& id : s->_cf_dirty | boost::adaptors::map_keys) {
```

Comment:
I don't understand this loop.

Let's start with documenting the algorithm in the code, and what `check_data_age` is supposed to do.
And probably rename it accordingly...

Reply:
See above.